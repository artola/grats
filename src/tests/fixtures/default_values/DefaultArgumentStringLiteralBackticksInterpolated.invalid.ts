const CONSTANT = "constant";

/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello({ greeting = `hello ${CONSTANT}` }: { greeting: string }): string {
    return `${greeting} world!`;
  }
}
