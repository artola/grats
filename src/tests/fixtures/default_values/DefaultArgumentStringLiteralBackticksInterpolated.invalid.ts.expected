-----------------
INPUT
----------------- 
const CONSTANT = "constant";

/** @gqlType */
export default class SomeType {
  /** @gqlField */
  hello({ greeting = `hello ${CONSTANT}` }: { greeting: string }): string {
    return `${greeting} world!`;
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/default_values/DefaultArgumentStringLiteralBackticksInterpolated.invalid.ts:6:22 - error: Expected GraphQL field argument default values to be a literal. Grats interprets argument defaults as GraphQL default values, which must be literals. For example: `10` or `"foo"`.

If you think Grats should be able to infer this constant value, please report an issue at https://github.com/captbaritone/grats/issues.

6   hello({ greeting = `hello ${CONSTANT}` }: { greeting: string }): string {
                       ~~~~~~~~~~~~~~~~~~~
